rotkehlchen/api/v1/schemas.py

```diff
@@ -3531,3 +3533,99 @@ class FalsePositveSpamTokenSchema(Schema):
 
 class SpamTokenSchema(Schema):
     token = AssetField(expected_type=EvmToken, required=True)
+
+
+class AnyBlockchainAddress(Schema):
+    """
+    Schema for two fields, one being the address field and the other the
+    blockchain where it belongs. The address can belong to any of the chains that we currently
+    support and it will check if the format is correct for the blockchain.
+    """
+    address = fields.String(load_default=None)
+    blockchain = BlockchainField(load_default=None)
+
+    @validates_schema
+    def validate_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        """Check that the combination of address and blockchaina re correct"""
+        if (data['address'] is None) ^ (data['blockchain'] is None):
+            raise ValidationError(
+                'If any of address or blockchain is provided both need to be provided',
+            )
+
+        if data['blockchain'] is None:
+            return
+
+        blockchain = cast(SupportedBlockchain, data['blockchain'])
+        if ((
+            blockchain == SupportedBlockchain.BITCOIN and
+            is_valid_btc_address(data['address']) is False
+        ) or (
+            blockchain == SupportedBlockchain.BITCOIN_CASH and
+            is_valid_bitcoin_cash_address(data['address']) is False
+        ) or (
+            blockchain.get_chain_type() == 'substrate' and
+            is_valid_substrate_address(chain=blockchain, value=data['address']) is False  # type: ignore  # expects polkadot or kusama
+        ) or (
+            blockchain.is_evmlike() and
+            to_checksum_address(data['address']) == data['address']
+        )):
+            raise ValidationError(
+                f'Given value {data["address"]} is not a {blockchain} address',
+                field_name='address',
+            )
+
+
+class CalendarCommonEntrySchema(AnyBlockchainAddress):
+    name = fields.String(required=True)
+    description = fields.String(load_default=None)
+    counterparty = EvmCounterpartyField()
+
+    def __init__(self, chain_aggregator: 'ChainsAggregator') -> None:
+        super().__init__()
+        self.declared_fields['counterparty'].set_counterparties(  # type: ignore
+            counterparties={x.identifier for x in chain_aggregator.get_all_counterparties()},
+        )
+
+
+class NewCalendarEntrySchema(CalendarCommonEntrySchema):
+    timestamp = TimestampField(required=True)
+
+
+class UpdateCalendarSchema(NewCalendarEntrySchema, IntegerIdentifierSchema):
+    ...
+
+
+class QueryCalendarSchema(DBPaginationSchema, TimestampRangeSchema, DBOrderBySchema):
+    name = fields.String(load_default=None)
+    description = fields.String(load_default=None)
+    counterparty = fields.String(load_default=None)
```

Comment:
here counterparty and addresses are still not checked afaics

Reply:
Had to add a bit of logic since to check the address you need the blockchain.